lms/djangoapps/course_home_api/course_metadata/v1/views.py

```diff
@@ -85,8 +85,19 @@ def get(self, request, *args, **kwargs):
         enrollment = CourseEnrollment.get_enrollment(request.user, course_key_string)
         user_is_enrolled = bool(enrollment and enrollment.is_active)
 
-        courseware_meta = CoursewareMeta(course_key, request, request.user.username)
-        can_load_courseware = courseware_meta.is_microfrontend_enabled_for_user()
```

Comment:
Creating a `CoursewareMeta` object does a bunch of stuff, including some db lookups, a modulestore lookup, and extra `has_access` calls. So I've replicated just the calls we care about below (`check_course_access` and `courseware_mfe_is_visible`)